### PR TITLE
22560-Clean-magic-numbers-in-InputEventHandler-subclasses

### PR DESCRIPTION
**Pull request by juliendelplanque**
Removed hard-coded constants (magic numbers) and replaced them by references to class variables holding the constants.

I think this change will improve readability of the code a lot.

### CODE REVIEW

File: src/System-VMEvents/EventSensorConstants.class.st [hunk 1]
```diff
@@ -40,15 +48,27 @@ EventSensorConstants class >> initialize [
...
 ]
 
 { #category : #'private - initialization' }
 EventSensorConstants class >> initializeButtonBits [
 
 	RedButtonBit := 4.
-	BlueButtonBit := 2.
-	YellowButtonBit := 1
+	BlueButtonBit := 1.
+	YellowButtonBit := 2.
```
Comment (juliendelplanque) [inline src/System-VMEvents/EventSensorConstants.class.st:60]: Here I realised that, compared to the constants used in the handlers, there was an inconsistency, the value for the blue and the yellow buttons were flipped here...

File: src/System-VMEvents/InputEventSensor.class.st [hunk 2]
```diff
@@ -115,24 +115,26 @@ InputEventSensor class >> swapMouseButtons [
...
-	^self modifiers anyMask: 16r0E	"cmd | opt | ctrl"
+	^self modifiers anyMask: ModifierKeysBitMask
 ]
 
 { #category : #mouse }
 InputEventSensor >> blueButtonPressed [
 	"Answer whether only the blue mouse button is being pressed. 
 	This is the third mouse button or cmd+click on the Mac."
 
-	^(self mouseButtons bitAnd: 7) = 1
```
Comment (juliendelplanque) [inline src/System-VMEvents/InputEventSensor.class.st:135]: Here you can see the inconsistency with the original shared pool class variable value.

File: src/System-VMEvents/InputEventSensor.class.st [hunk 3]
```diff
@@ -576,6 +594,6 @@ InputEventSensor >> yellowButtonPressed [
 	"Answer whether only the yellow mouse button is being pressed. 
 	This is the second mouse button or option+click on the Mac."
 
-	^(self mouseButtons bitAnd: 7) = 2
```
Comment (juliendelplanque) [inline src/System-VMEvents/InputEventSensor.class.st:579]: Here you can see the other inconsistency with the original shared pool class variable value.

**Comment by jecisc** [comment]
I tested the resulting image and it seems to work fine.

**Review by jecisc — APPROVED** [review]
Changes seem good to me. The only thing I don't know of is if it will impact perfs.

**Comment by MarcusDenker** [comment]
it is "pushLiteralVariable" vs. "pushLiteral" byte code.. should not make much difference.

**Comment by juliendelplanque** [comment]
Well, for me the difference is not feel-able if it exists which is the most important. I didn't felt one at least.

But the code understandability is far better with this change.

EDIT: But indeed, my feeling about the difference is not really important. If someone can make a benchmark I'm interested but personally I do not have the skills.

**Comment by MarcusDenker** [comment]
this PR was done against a version were the bootstrap was broken. As the PR check uses this version, the PR test check stays broken even though we fixed the broken build in the meantime.

Solution: update the PR to be against the latest version.

**Comment by juliendelplanque** [comment]
@MarcusDenker how can I do that?

**Comment by MarcusDenker** [comment]
No idea

I am not even sure why the PR is checked not against the latest version. But if I re-tun, it get he bootstrap error.

**Comment by juliendelplanque** [comment]
We should look together because it was a pain to refactor this code. I hope it will not be cancelled.

**Comment by jecisc** [comment]
Maybe merging dev into it can correct the problem? (Just an idea)

**Comment by guillep** [comment]
Just need to close the PR and reopen it.

**Comment by juliendelplanque** [comment]
Thank you @guillep :-)

**Comment by juliendelplanque** [comment]
Coming back on this PR, should I do something?

**Pull request merged**